conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -228,6 +231,14 @@ public static JavaFile generateAliasType(
                 .build();
     }
 
+    private static CodeBlock getStaticConstructor(Type type, ClassName thisClass, Options options) {
+        if (options.defensiveCollectionAliases() && type.accept(MoreVisitors.IS_COLLECTION)) {
+            return type.accept(new DefensiveCollectionConstructorVisitor(thisClass, options))
+                    .orElseThrow();
+        }
+        return CodeBlock.of("return new $T(value)", thisClass);
```

Comment:
Thoughts on moving the options check into `DefensiveCollectionConstructorVisitor` where we implicitly handle the `MoreVisitors.IS_COLLECTION` check via the visitor? We'd probably want to rename `DefensiveCollectionConstructorVisitor` to `AliasFactoryVisotor implements Visitor<CodeBlock>`